### DISCUSSION TITLE
Video Capturing (zmbv)

### DISCUSSION THREAD

**Discussion by GiulianoPastorino** [dosbox-staging/dosbox-staging#3825] [Q&A] [answered]
Hi everyone,

DOSBox Staging's \doc\manual.txt informs us that:

`Ctrl+F7       Start/Stop recording video output to a zmbv file.`

When I used these keys, DOSBox Staging created a .AVI file in the \capture folder. But when I play the file, it plays a black screen with no sound.

What am I doing wrong?

Cheers.

**Comment by MasterO2** [comment, marked as answer]
@GiulianoPastorino, what media player did you use to play the `.avi` file?  

[SMPlayer](https://www.smplayer.info/), [VLC](https://www.videolan.org/vlc/), and many other open-source media players can play that file format without issues.

I would recommend trying any of the open-source media players that I just mentioned to verify if that solved your issue.

**Reply by interloper98** [reply]
@GiulianoPastorino , you can also use tools like HandBrake to scale the `.avi` to aspect-correct 4:3 resolution (like 640x480 or 960x720), and encode it to a format that's common on the web, like h264+aac, or av1+opus.

Once it's in a web-compatible format, most hardware players like phones & TVs should be able to play it, too.

**Reply by GiulianoPastorino** [reply]
@MasterO2, I was using Windows Media Player. I've downloaded SMPlayer as you instructed, and it works! Kinda... At my first attempt, SMPlayer played the video, but still no sound. I then realized that I was playing the game using MIDI (via Munt). So for my second attempt I've changed the game config to Soundblaster, and now I can record video with sound. Thank you!

It would be lovely if we could also record MIDI sounds...

@interloper98, thank you for this advice! I will definitely try HandBrake.

Cheers.

**Reply by johnnovak** [reply]
@GiulianoPastorino Any specific reason for using Munt as an external process when we have it built-in? It's exactly the same Munt, just you don't need the virtual MIDI cables, you can configure the ROM version per game (per config), we use slightly better default settings than stock Munt to eliminate some game issues, and then the MT-32 output would be recorded with your video.

And yes, the ZMBV encoded AVI captures are just an intermediate format that very few software can play. People are expected to convert it into some universal format with HandBrake or ffmpeg, these can read ZMBV.

**Reply by GiulianoPastorino** [reply]
@johnnovak, thank you for your advice!

I prefer to use Munt because I like to read the MIDI messages as they originaly appeared in a Roland MT-32, as shown below:

![image](https://github.com/user-attachments/assets/a43d1e2f-d1c1-4207-a3ef-93a36871f95a)

But I will follow your advice and use the built-in process, so I can record the sound output.

Thank you!

Cheers.

**Reply by johnnovak** [reply]
@GiulianoPastorino We are planning to introduce MT-32 _and_ SC-55 message overlays to Staging, probably next year. I like those messages too 😎. Enjoy Staging! 😄
